interfaces/builtin/broadcom_asic_control.go

```diff
@@ -0,0 +1,127 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/kmod"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const broadcomAsicControlSummary = `allows using the broadcom-asic kernel module`
+
+const broadcomAsicControlBaseDeclarationSlots = `
+  broadcom-asic-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+const broadcomAsicControlConnectedPlugAppArmor = `
+# Description: Allow access to broadcom asic kernel module.
+
+/sys/module/linux_kernel_bde/initstate r,
+/sys/module/linux_user_bde/initstate r,
+/sys/module/linux_kernel_bde/holders/ r,
+/sys/module/linux_user_bde/holders/ r,
+/sys/module/linux_user_bde/holders/** r,
+/sys/module/linux_user_bde/refcnt r,
+/sys/module/linux_bcm_knet/initstate r,
+/sys/module/linux_bcm_knet/holders/ r,
+/sys/module/linux_bcm_knet/refcnt r,
```

Comment:
Since these are all only reads, I think the above can all be collapsed into (which is easier to read and futureproof):
```
/sys/module/linux_bcm_knet/{,**} r,
/sys/module/linux_kernel_bde/{,**} r,
/sys/module/linux_user_bde/{,**} r,
```